internal/admission/admission.go

```diff
@@ -59,18 +58,38 @@ func (r *Router) InjectDecoder(d *admission.Decoder) error {
 	if d == nil {
 		return fmt.Errorf("decoder cannot be nil")
 	}
-	for _, h := range r.handlers {
-		if err := h.InjectDecoder(d); err != nil {
-			return err
+	for _, hs := range r.handlers {
+		for _, h := range hs {
+			if err := h.InjectDecoder(d); err != nil {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
 func (r *Router) Handle(ctx context.Context, req admission.Request) admission.Response {
-	h, ok := r.handlers[req.RequestKind.Kind]
+
+	kind := req.RequestKind
```

Comment:
Do we know if we're planning to use [matchPolicy](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-matchpolicy) `Exact` or `Equivalent`?

If we'll be using `Equivalent` then I think we should use `req.Kind`, since we'd be getting the converted request.